--- busybox-1.38.0.ori/networking/httpd.c	2026-06-09 15:21:19.058013858 +0000
+++ busybox-1.38.0/networking/httpd.c	2026-06-09 15:21:20.426057792 +0000
@@ -2260,8 +2260,12 @@
 			goto end_check_passwd;
 		}
  bad_input:
-		/* Comparing plaintext "user:pass" in one go */
-		r = strcmp(cur->after_colon, user_and_passwd);
+		/* Check if auth is disabled for this path */
+		r = strcmp(cur->after_colon, ":");
+		if (r != 0) {
+			/* Comparing plaintext "user:pass" in one go */
+			r = strcmp(cur->after_colon, user_and_passwd);
+		}
  end_check_passwd:
 		if (r == 0) {
 			remoteuser = xstrndup(user_and_passwd,
